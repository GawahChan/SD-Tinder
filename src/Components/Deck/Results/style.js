import styled from 'styled-components';

export const ResultsContainer = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    min-height: 100vh;

    text-align: center;
    margin: 30px;
`

export const ResultDisplay = styled.div`
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
`

export const ResultHeader = styled.h1`
    color: white;
    font-weight: 400;
    font-size: 2.5rem;
`

export const ResultTitle = styled(ResultHeader)`
    font-size: 1.5rem;
`

export const ResultSubTitle = styled(ResultHeader)`
    font-size: 1rem;
    margin: 0;
`

export const ResultProfile = styled.div`
    background-image: ${props => {
        return (
            `linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0), rgba(0,0,0,0.6), rgba(0,0,0,1)),
            url(${require(`../../../Common/Profile/${props.img}.jpg`)})     `
        )
    }};
    
    border-radius: 10px;
    box-shadow: 1px 1px 8px #333;
    margin: 10px;

    background-size: cover;
    background-position: top;
    overflow: hidden;

    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    
    min-width: 250px;
    min-height: 250px;
`