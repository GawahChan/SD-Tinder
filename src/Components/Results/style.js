import styled from "styled-components";

export const ResultsPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;

  min-height: 100vh;

  text-align: center;
  box-sizing: border-box;
  padding: 30px;
`;

export const ResultsContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  box-sizing: border-box;
  padding: 20px;
  flex: 1;
`;

export const ResultContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  box-sizing: border-box;
  padding: 10px;
  flex: 1;
`;

export const ResultDisplay = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-items: center;
  flex: 1;
`;

export const ResultProfile = styled.div`
  background-image: ${props => {
    return `linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0), rgba(0,0,0,0.2), rgba(0,0,0,1)),
            url(${require(`../../Common/Profile/${props.img}.jpg`)})     `;
  }};

  border-radius: 10px;
  border: ${props => props.theme.border};
  box-shadow: ${props => props.theme.shadow};
  margin: 10px;

  background-size: cover;
  background-position: top;
  overflow: hidden;

  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  min-width: 250px;
  min-height: 250px;
`;

export const Container = styled.div`
  box-sizing: border-box;
  padding: 10px;
`;
