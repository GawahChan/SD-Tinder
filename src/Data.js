const Data = [
    {id: 0, name: 'Mila Kunis', age: '30', location: 'LA', bio: 'Actress' }
];

export default Data;