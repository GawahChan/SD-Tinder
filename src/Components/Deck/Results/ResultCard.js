import React from "react";
import PropTypes from "prop-types";
import { Title } from "../../../Common/Styled/Typography.styles";
import { ResultProfile, Container } from "./style";

function ResultCard({ img, name }) {
  return (
    <ResultProfile img={img}>
      <Container>
        <Title>{name}</Title>
      </Container>
    </ResultProfile>
  );
}

ResultCard.propTypes = {
  img: PropTypes.string,
  name: PropTypes.string
};

export default ResultCard;
