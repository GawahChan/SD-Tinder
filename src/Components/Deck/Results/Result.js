import React from "react";
import ResultCard from "./ResultCard";
import { Title, Subtitle } from "../../../Common/Styled/Typography.styles";
import { ResultDisplay, Container } from "./style";

function Result(props) {
  return (
    <Container>
      <Title>{props.title}</Title>
      <ResultDisplay>
        {props.stuntDouble.length > 0 ? (
          props.stuntDouble.map(stuntDouble => {
            return (
              <ResultCard
                key={stuntDouble.id}
                img={stuntDouble.img}
                name={stuntDouble.name}
              />
            );
          })
        ) : (
          <Subtitle>{props.subTitle}</Subtitle>
        )}
      </ResultDisplay>
    </Container>
  );
}

export default Result;
