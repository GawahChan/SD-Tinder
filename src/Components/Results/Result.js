import React from "react";
import PropTypes from "prop-types";
import { ThemeProvider } from "styled-components";
import { theme } from "../../Theme";
import ResultCard from "./ResultCard";
import { Title, Subtitle } from "../../Common/Styled/Typography.styles";
import { ResultDisplay, ResultContainer } from "./style";

function Result({ id, title, stuntDouble, subTitle }) {
  return (
    <ThemeProvider theme={theme[id]}>
      <ResultContainer>
        <Title>{title}</Title>
        <ResultDisplay>
          {stuntDouble.length > 0 ? (
            stuntDouble.map(stuntDouble => {
              return (
                <ResultCard
                  key={stuntDouble.id}
                  img={stuntDouble.img}
                  name={stuntDouble.name}
                />
              );
            })
          ) : (
            <Subtitle>{subTitle}</Subtitle>
          )}
        </ResultDisplay>
      </ResultContainer>
    </ThemeProvider>
  );
}

Result.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  stuntDouble: PropTypes.array,
  subTitle: PropTypes.string
};

export default Result;
