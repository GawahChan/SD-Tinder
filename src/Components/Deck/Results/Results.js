import React from 'react';
import Result from './Result';

import { ResultsContainer, ResultHeader } from './style';

function Results(props) {
    return (
        <ResultsContainer>
            <ResultHeader>There's no one new around you</ResultHeader>
            <Result
                title={"People you superliked!"}
                stuntDouble={props.superliked}
                subTitle={"You didn't superlike anyone"}
            />
            <Result
                title={"People you liked... let's hope they like you too!"}
                stuntDouble={props.liked}
                subTitle={"You didn't like anyone"}
            />
            <Result 
                title={"People you disliked"}
                stuntDouble={props.disliked}
                subTitle={"Cast a wide net they said..."}
                />
        </ResultsContainer>
    );
}

export default Results;