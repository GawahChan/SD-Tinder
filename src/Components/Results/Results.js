import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { resetAllStuntDoubles } from "../../Store/Actions/actionTypes";

import Result from "./Result";
import { DECK_URL } from "../../Constants";
import { Header } from "../../Common/Styled/Typography.styles";
import { HomeLink, ResultButton } from "../../Common/Styled/Buttons.styles";
import { ResultsPageContainer, ResultsContainer, Container } from "./style";

class Results extends Component {
  render() {
    const {
      stuntDoublesList,
      superliked,
      liked,
      disliked,
      resetAllStuntDoubles
    } = this.props;

    const showResult =
      superliked.length > 0 || liked.length > 0 || disliked.length > 0;

    const HeaderText =
      stuntDoublesList.length > 0
        ? `You never know who you might meet...`
        : `There's no one new around you`;

    const ButtonText =
      stuntDoublesList.length === 0
        ? `Reset all profiles`
        : showResult
        ? "Carry on swiping"
        : `Start swiping now!`;

    return (
      <ResultsPageContainer>
        <Container>
          <Header>{HeaderText}</Header>
          <Container>
            {stuntDoublesList.length > 0 ? (
              <HomeLink to={DECK_URL}>
                <ResultButton>{ButtonText}</ResultButton>
              </HomeLink>
            ) : (
              <HomeLink to={DECK_URL} onClick={() => resetAllStuntDoubles()}>
                <ResultButton>{ButtonText}</ResultButton>
              </HomeLink>
            )}
          </Container>
        </Container>
        {showResult && (
          <ResultsContainer>
            <Result
              id="superliked"
              title={"People you superliked!"}
              stuntDouble={superliked}
              subTitle={"You didn't superlike anyone..."}
            />
            <Result
              id="liked"
              title={"People you liked... let's hope they like you too!"}
              stuntDouble={liked}
              subTitle={"You didn't like anyone..."}
            />
            <Result
              id="disliked"
              title={"People you disliked"}
              stuntDouble={disliked}
              subTitle={"Cast a wide net they said..."}
            />
          </ResultsContainer>
        )}
      </ResultsPageContainer>
    );
  }
}

Results.propTypes = {
  stuntDoubleList: PropTypes.array,
  superliked: PropTypes.array,
  liked: PropTypes.array,
  disliked: PropTypes.array,
  resetAllStuntDoubles: PropTypes.func
};

const mapStateToProps = state => {
  return {
    stuntDoublesList: state.stuntDoubles,
    superliked: state.superliked,
    liked: state.liked,
    disliked: state.disliked
  };
};

const mapDispatchToProps = dispatch => {
  return bindActionCreators({ resetAllStuntDoubles }, dispatch);
};
export default connect(mapStateToProps, mapDispatchToProps)(Results);
