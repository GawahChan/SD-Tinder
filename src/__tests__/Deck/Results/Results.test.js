import React from 'react';
import { shallow } from 'enzyme';
import Results from '../../../Components/Deck/Results/Results';
import Data from '../../../Data';

describe('Result component', () => {
    let props = { liked: [Data[0]], superliked: [Data[1]], disliked: [Data[2]]}
    let result = shallow(<Results {...props}/>);

    it('renders correctly', () => {
        expect(result).toMatchSnapshot();
    });

    it('renders <Result> once for each prop', () => {
        expect(result.children().length).toEqual(4);
        expect(result.find('Result').length).toEqual(3);
    });

    // describe('Result component where all props have no data', () => {
    //     let props = { liked: [], superliked: [], disliked: []}
    //     let result = shallow(<Results {...props}/>);

    //     it('does not render <ResultCard/> for .result-propiles.superliked', () => {
    //         expect(result.find('.result-profiles.superliked').children('ResultCard').length).toEqual(0);
    //         expect(result.find('.result-profiles.superliked').children('h3').length).toEqual(1);
    //     });
    //     it('does not render <ResultCard/> for .result-propiles.liked', () => {
    //         expect(result.find('.result-profiles.liked').children('ResultCard').length).toEqual(0);
    //         expect(result.find('.result-profiles.liked').children('h3').length).toEqual(1);
    //     })
    //     it('does not render <ResultCard/> for .result-propiles.disliked', () => {
    //         expect(result.find('.result-profiles.disliked').children('ResultCard').length).toEqual(0);
    //         expect(result.find('.result-profiles.disliked').children('h3').length).toEqual(1);
    //     })
    // });








})