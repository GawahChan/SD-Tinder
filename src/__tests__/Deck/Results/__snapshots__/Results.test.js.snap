// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result component renders correctly 1`] = `
<styled.div>
  <styled.h1>
    There's no one new around you
  </styled.h1>
  <Result
    stuntDouble={
      Array [
        Object {
          "age": "58",
          "bio": "Just finished working as a stunt performer in the movie Dhaka and am looking for a new role. I've stunt doubled for many characters in both the Marvel and DC universe. Drop me a message if you're interested",
          "id": 1,
          "img": "Richard-Cetrone",
          "location": "Pennsylvania, USA",
          "name": "Richard Cetrone",
        },
      ]
    }
    subTitle="You didn't superlike anyone"
    title="People you superliked!"
  />
  <Result
    stuntDouble={
      Array [
        Object {
          "age": "28",
          "bio": "Recently finished stunt doubling for the movie 'Alita: Battle Angel' and am looking for a new role! Have stunt doubled for major films such as Wonder Woman (Gal Gadot), 'The Punisher' and 'Marvel's Agents of S.H.I.E.L.D'. If you're interested, swipe right and drop me a message!",
          "id": 0,
          "img": "Alicia-Vela-Bailey",
          "location": "LA, USA",
          "name": "Alicia Vela-Bailey",
        },
      ]
    }
    subTitle="You didn't like anyone"
    title="People you liked... let's hope they like you too!"
  />
  <Result
    stuntDouble={
      Array [
        Object {
          "age": "32",
          "bio": "From stunt doubling as the champion of the 74th Hunger Games to multiple superheroines from the Marvel Universe, I'm looking for a new action packed role! Let me know if you're interested!",
          "id": 2,
          "img": "Renae-Moneymaker",
          "location": "LA, USA",
          "name": "Renae Moneymaker",
        },
      ]
    }
    subTitle="Cast a wide net they said..."
    title="People you disliked"
  />
</styled.div>
`;
